yarpcerrors/errors.go

```diff
@@ -49,6 +49,9 @@ func FromError(err error) *Status {
 	if status, ok := err.(*Status); ok {
 		return status
 	}
+	if statusImpl, ok := err.(interface{ YARPCError() *Status }); ok {
```

Comment:
can we also add this to `IsStatus` and update the documentation on both to indicate that we look for this interface?